### ISSUE TITLE
When Blockman extension enabled, text selection is invisible

### ISSUE THREAD

**Issue by leodevbro** [getcursor/cursor#1666]
Here you can see the full description and screenshots too:

https://github.com/leodevbro/vscode-blockman/issues/141

In short, VS Code does not have this issue, only Cursor editor (VS Code fork) has it.

**Comment by SaidTorres3** [comment]
Main reason to not use cursor right now

**Comment by ftsmasaki** [comment]
I would love to be able to use Blockman with Cursor!

**Comment by finscn** [comment]
Me too

**Comment by YoraiLevi** [comment]
+1

**Comment by Jospin-Dk** [comment]
Same
A fix for this would bring back to cursor!

**Comment by krazykriskomar** [comment]
+1 to this. I have a number of people on my team who use Blockman. It'd be nice to be able to use it again.

**Comment by mm-zacharydavison** [comment]
+1

**Comment by FixedBit** [comment]
I am literally here right now to find an answer and without one I am going to have to wait on using this editor.

**Comment by leodevbro** [comment]
I think at least I diagnosed the problem.

Steps to see that the diagnose is correct:
* Render some blocks with Blockman (in Cursor editor). We see that text selection is not visible.
* Press `F1` and find `Developer: Toggle Developer Tools`, click on it to open Chrome (or Chrome-like) dev tools.
* Go to `Sources` tab -> `Page` subtab -> `top` -> `Electron Isolated Context` -> `c:/Users/LeoKa/AppData/Local/Programs/cursor/resources/app` (yours will be different) -> `out` -> `vs` -> `workbench` -> `workbench.desktop.main.css` file.
* Search `.monaco-editor .view-line>span` selector.
* Comment out `z-index: 4` style.
* Now we should see that text selection is visible.

![Image](https://github.com/user-attachments/assets/d7624dd3-8354-41d7-87f5-f3af2e68c1cb)

This `z-index` causes this problem because the layering structure of these three things: texts, VSCode selection effects and `vscode.window.createTextEditorDecorationType` elements (Which is the elements that Blockman renders) is managed by [Stacking without the z-index property](Stacking without the z-index property) (z-index is `auto` in most of the elements of the structure), so if anyone creates a fork-editor (like Cursor) and applies z-index anywhere in this structure, it should be done very carefully, because it can possible break the layering order, and I guess the Cursor editor team made this exact mistake. So, maybe they can fix this pretty easily.

**Comment by saadfarooq07** [comment]
+1

**Comment by leodevbro** [comment]
Before the Cursor team fixes this problem, you can use this 10 second manual fix method which seems working fine and you just need to do it after every restart of Cursor editor.

10 second steps:
* Press `F1` and find `Developer: Toggle Developer Tools`, click on it to open Chrome (or Chrome-like) dev tools.
* Paste and run this code in the Console:
```js
// maybe first it requires to type: allow pasting

const manualFixOfCursorLineDecorationZIndex = () => {
  // Create a Trusted Type policy using an allowed policy name
  const escapeHTMLPolicy = trustedTypes.createPolicy("safeInnerHtml", {
    createHTML: (to_escape) => to_escape,
  });

  // Use the policy to set innerHTML
  const style = document.createElement("style");
  style.innerHTML = escapeHTMLPolicy.createHTML(`
    .monaco-editor .view-line > span {
      z-index: initial; // if it does not work, try with !important
    }
  `);
  document.head.appendChild(style);
};

manualFixOfCursorLineDecorationZIndex();

```

![Image](https://github.com/user-attachments/assets/d2ca0b73-7810-474c-ad75-d2d71e564bfa)

**Comment by ShaysFrame** [comment]
Do you know if this will ever be fixed?

**Comment by leerob** [comment]
Thanks for opening this issue! We've moved our community support to the [Cursor Forum](https://forum.cursor.com) where you can get help from both the team and other users.

If this is a specific bug that you can still reproduce, please open a new issue with clear reproduction steps on the forum.

Thanks you!
